network/views.py

```diff
@@ -55,23 +62,43 @@ def post_detail(request, pk):
     #TODO: implement LIKE 
 
 
-@login_required
-def following(request):
-    # Query for requested data
+def follow_list(request, username, fol_type):
+
+    # Query for requested user
     try: 
-        obj = UserFollowing.objects.filter(user=request.user.id)
-    except UserFollowing.DoesNotExist:
-        return JsonResponse({"error": f"User id {pk} not found."}, status=404)
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        return JsonResponse({"error": f"User {username} not found."}, status=404)
 
-    # Return following list
-    following_list = []
+    # Get follow list
+    follow_list = []
     if request.method == 'GET':
-        for i in range(len(obj)):
-            following_list.append(obj[i].following_user.username)
-        data = {
-            "user": request.user.username,
-            "following": following_list
-        }
+
+        if fol_type == 'following':
+            fols = user.following.all()
```

Comment:
i think you can just do return 
{
user: username
"following": [user.username for user.following_list()]
} 

and similar for the one below